ls-ecaps.c

```diff
@@ -690,30 +690,202 @@ cap_rcec(struct device *d, int where)
 }
 
 static void
-cap_dvsec_cxl(struct device *d, int where)
+cxl_range(u64 base, u64 size, int n)
 {
-  u16 l;
+  u32 interleave[] = { 0, 256, 4096, 512, 1024, 2048, 8192, 16384 };
+  const char *type[] = { "Volatile", "Non-volatile", "CDAT" };
+  const char *class[] = { "DRAM", "Storage", "CDAT" };
+  u16 w;
 
-  printf(": CXL\n");
-  if (verbose < 2)
-    return;
+  w = (u16) size;
 
-  if (!config_fetch(d, where + PCI_CXL_CAP, 12))
+  size &= ~0x0fffffffULL;
+
+  printf("\t\tRange%d: %"PRIx64"-%"PRIx64"\n", n, base, base + size - 1);
```

Comment:
Just a small suggestion, what about adding left padding by zeros for range numbers? Like when printing base/limit numbers for BARs and windows.